File: mysqlclient-practices/basic/insert_test.py
# from MySQLdb import connect, OperationalError
#
# try:
#     # 1. 연결
#     db=connect(
#         user='webdb',
#         host='127.0.0.1',
#         port= 3306,
#         db= 'webdb',
#         charset= 'utf8')
#
#     # 2.
#     cursor = db.cursor()
#
#     # 3. sql()
#     sql= "insert into pet values('소금이','여혜린','cat','m','2014-04-20',null)"
#     count = cursor.execute(sql)
#
#     # 4. commit
#     db.commit()
#
#
#     # 5. 자원 정리
#     cursor.close()
#     db.close()
#
#
#     # 결과 확인
#     print(f '실행 결과: {"성공" if count == 1 else "실패"}')
#
#     # 에러 처리
# except OperationalError as e:
#     print(f '에러:{e}')

from MySQLdb import connect, OperationalError

try:
    # 1. 연결
    db = connect(
        user='webdb',
        password='webdb',
        host='127.0.0.1',
        port=3306,
        db='webdb',
        charset='utf8')

    # 2. cursor 생성
    cursor = db.cursor()

    # 3. sql(insert 문) 실행
    sql = "insert into pet values('소금이','여혜린','cat','m','2014-04-20',null)"
    count = cursor.execute(sql)

    # 4. commit
    db.commit()

    # 5. 자원 정리
    cursor.close()
    db.close()

    # 결과 확인
    print(f'실행 결과: {"성공" if count == 1 else "실패"}')

except OperationalError as e:
    # 에러 처리
    print(f'에러: {e}')

File: JDBC_practices/src/main/java/testself/InsertTest.java
package testself;

import java.beans.Statement;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;

public class InsertTest {

	public static void main(String[] args) {
		insert("시스템");
		insert("마켓팅");
		insert("운영");
		

	}

	public static boolean main(String[] args) {
		boolean resilt =false;
		
	Connection conn = null;
	Statement stmt=null;
	
	try {
		// 1. JDBC Drive Class Loading
		Class.forName("org.mariadb.jdbc.Driver");

		// 2.연결하기
		String url = "jdbc:mysql://127.0.0.1:3306/webdb?charset=utf8";
		conn = DriverManager.getConnection(url, "webdb", "webdb");
		//3. Statement 생성
		 stmt =conn.createStatement();
		
		//4.실행
		String sql=
				"insert"+
				"into dept"+
				"values(null,'디자인')";

		int count= stmt.executeUpdate(sql);
				
		return resilt;
		
	} catch (ClassNotFoundException e) {
		System.out.println("드라이버로딩 실패:" + e);
	} catch (SQLException e) {
		System.out.println("Error:" + e);
	} finally {
		try {
			if (stmt != null) {
				stmt.close();
			}if (conn != null) {
				conn.close();
			}
		} catch (SQLException e) {
			e.printStackTrace();
		}

	}
}
}
